### PR TITLE
Adds support for FCM V1

### PR DESCRIPTION
**Pull request by aziztitu**
Things to consider before you submit the PR:

* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [ ] Did you add unit tests?
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

Currently, this SDK registers the devices as FCM Legacy registrations/installations on ANH. Google will stop supporting FCM Legacy HTTP on June 20, 2024. 

This PR adds support for registering new devices as FCM V1 registrations/installations and migrating existing FCM registrations/installations to FCM V1.

### CODE REVIEW

File: notification-hubs-sdk/src/main/java/com/microsoft/windowsazure/messaging/notificationhubs/Installation.java [hunk 1]
```diff
@@ -13,6 +13,7 @@
  */
 public class Installation implements Taggable {
     private String mPushChannel;
+    private String mPlatform = "fcm";
```
Comment (marstr) [inline notification-hubs-sdk/src/main/java/com/microsoft/windowsazure/messaging/notificationhubs/Installation.java:16]: So by default its "fcm" which will map to the Legacy HTTP API platform. Customers looking to use the new FCM V1 API platform will need to add a line of code to change this?

In a future version will we change the default behavior to be FCM V1?

File: notification-hubs-sdk/src/fcm/java/com/microsoft/windowsazure/messaging/NotificationHubLegacyExtension.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+import androidx.annotation.NonNull;
+
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.messaging.FirebaseMessaging;
+
+public class NotificationHubLegacyExtension {
+    private static final String MIGRATED_TO_FCM_V1_KEY = "__ANH__MIGRATED_TO_FCM_V1";
+
+    public static Task<Void> migrateToFcmV1(Context context, NotificationHub hub, String ...tags) {
```
Comment (marstr) [inline notification-hubs-sdk/src/fcm/java/com/microsoft/windowsazure/messaging/NotificationHubLegacyExtension.java:18]: I love it - are we going to deprecate this after July when Google has truly turned the servers off of the legacy FCM APIs?

File: notification-hubs-sample-app-java/src/main/java/com/example/notification_hubs_sample_app_java/legacy/LegacyNotificationHubManager.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+    public static LegacyNotificationHubManager getInstance() {
+        if (sInstance == null) {
+            sInstance = new LegacyNotificationHubManager();
+        }
+        return sInstance;
+    }
+
+    public void initialize(Context context) {
+        HttpsURLConnection.setDefaultHostnameVerifier((s, sslSession) -> true); // TODO: Remove after testing
```
[the remark this answers is not in the document]
Comment (aziztitu) [reply notification-hubs-sample-app-java/src/main/java/com/example/notification_hubs_sample_app_java/legacy/LegacyNotificationHubManager.java:40]: Nope, good catch!

File: notification-hubs-sdk/src/main/java/com/microsoft/windowsazure/messaging/notificationhubs/Installation.java (hunk 1 above)
Comment (aziztitu) [reply notification-hubs-sdk/src/main/java/com/microsoft/windowsazure/messaging/notificationhubs/Installation.java:16]: Yes, that is correct. We are planning to release a new Minor version of the SDK (and keep "fcm" as the default platform). After Google deprecates FCM Legacy, we can release a new version that defaults the platform to "fcmV1".

Alternatively, we could also release a new Major version instead, and default the platform to "fcmV1". Do you have a preference?

File: notification-hubs-sdk/src/fcm/java/com/microsoft/windowsazure/messaging/NotificationHubLegacyExtension.java (hunk 2 above)
Comment (aziztitu) [reply notification-hubs-sdk/src/fcm/java/com/microsoft/windowsazure/messaging/NotificationHubLegacyExtension.java:18]: Yes, that is correct.

Comment (aziztitu) [reply notification-hubs-sdk/src/fcm/java/com/microsoft/windowsazure/messaging/NotificationHubLegacyExtension.java:18]: Note that this is only needed for customers using the Registration flow

File: notification-hubs-sdk/src/main/java/com/microsoft/windowsazure/messaging/notificationhubs/Installation.java (hunk 1 above)
Comment (marstr) [reply notification-hubs-sdk/src/main/java/com/microsoft/windowsazure/messaging/notificationhubs/Installation.java:16]: I like the planned you laid out, minor version now, major version 6 months from now.

**Review by jessHuh — APPROVED** [review]

**Review by jessHuh — APPROVED** [review]

**Pull request merged**